kong/plugins/datadog/handler.lua

```diff
@@ -0,0 +1,32 @@
+local log = require "kong.plugins.datadog.log"
+local BasePlugin = require "kong.plugins.base_plugin"
+local basic_serializer = require "kong.plugins.log-serializers.basic"
+
+local ngx_now = ngx.now
+
+local DatadogHandler = BasePlugin:extend()
+
+function DatadogHandler:new()
+  DatadogHandler.super.new(self, "datadog")
+end
+
+function DatadogHandler:body_filter(conf)
+  DatadogHandler.super.body_filter(self)
+  
+  ngx.ctx.datadog = {}
```

Comment:
This will reset the `ctx.datadog` table every time the body filter is being executed. (body filter) can be executed more than once during the same response). Better to create it somewhere else, like the [ALF serializer does](https://github.com/Mashape/kong/blob/next/kong/plugins/mashape-analytics/handler.lua#L62).